packages/rocketchat-language-plugin/server/add_or_update_user.js

```diff
@@ -0,0 +1,26 @@
+Meteor.methods({
+	add_or_update_user(channel, plugin_name) {
+		const channel_exist = Meteor.users.findOne({
+			$and: [{ _id: Meteor.userId()}, {'automatic_channels.name': channel}]
+		});
+		if (!channel_exist) {
+			Meteor.users.update(Meteor.userId(), {
+				$push: {
+					automatic_channels: {'name': channel, 'plugin': plugin_name, 'added_to':'true', 'blacklisted': 'false'}
+				}
+			});
+		} else {
+			Meteor.users.update({
+				_id: Meteor.userId(),
+				'automatic_channels.name': channel
+			}, {
+				$set: {
+					'automatic_channels.$.added_to' :'true'
```

Comment:
So it seems the solution is that we keep subdocuments for blacklisted channels around and do not remove them, no? So if a subdocument does not blacklisted set, we remove it if user changes the language. And if it is the same, we keep it and add a new subdocument for the new language.